File: 1.StrategyPattren/C++/ClassExample/ClassExample/MiniDuckSimulator.cpp
#include<iostream>
using namespace std;

#include "Duck.h"
#include "MallardDuck.h"
#include "RubberDuck.h"
#include "FlyRocketPowered.h"

int main()
{
	Duck *mallard = new MallardDuck();
	mallard->performFly();

	Duck *rubber = new RubberDuck();
	rubber->performFly();

	rubber->setFlyBehavior(new FlyRocketPowered()); // 실행 중 행위 변경
	rubber->performFly();

	return 0;
}

File: 1.StrategyPattren/JAVA/src/classexample/MiniDuckSimulator.java
package classexample;

public class MiniDuckSimulator {

	public static void main(String[] args) {

		Duck mallard = new MallardDuck();
		mallard.quack();
		mallard.performFly();
		
		Duck rubber = new RubberDuck();
		rubber.quack();
		rubber.performFly();

		rubber.setFlyBehavior(new FlyRocketPowered()); // 실행 중 행위 변경
		rubber.performFly();

	}
}
